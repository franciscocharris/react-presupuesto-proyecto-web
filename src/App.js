import React, { useState } from 'react'
import { ControlPresupuesto } from './components/ControlPresupuesto';
import { Formulario } from './components/Formulario';
import { ListaGastos } from './components/ListaGastos';
import { Pregunta } from './components/Pregunta'

export const App = () => {

    const [presupuesto, setPresupuesto] = useState(0);
    const [restante, setRestante] = useState(0);
    const [gastos, setGastos] = useState([]);

    // cuando hay nuevo gasto
    const agregarNuevoGasto = gasto => {
        setGastos([
            ...gastos,
            gasto
        ]);

        // restar del presupuesto (actualizar restante)
        setRestante(restante - gasto.cantidad);
    }

    // un state para la carga condiconal de componentes
    const [showPregunta, setShowPregunta] = useState(true);

    return (
        <div className='container'>
            <header>
                <h1>Gasto Semanal</h1>
                <a target='_blank'  rel="noreferrer"  href='https://www.facebook.com/profile.php?id=100013541311546' className='credito'> - por: Francisco M. Charris C. -</a>

                <div className='contenido-principal contenido'>
                    {
                        showPregunta
                        ?
                        // los parentesis hacen el return implicito
                        (
                            <Pregunta
                                setPresupuesto={setPresupuesto}
                                setRestante={setRestante}
                                setShowPregunta={setShowPregunta}
                            />
                        )
                        :
                        (
                            <div className='row'>
                                <div className='one-half column'>
                                    <Formulario  
                                       agregarNuevoGasto={agregarNuevoGasto} 
                                    />
                                </div>
                                <div className='one-half column'>
                                    <ListaGastos
                                        gastos={gastos}
                                    />

                                    <ControlPresupuesto 
                                        presupuesto={presupuesto}
                                        restante={restante}
                                    />
                                </div>
                            </div>
                        )
                    }

                </div>
            </header>
        </div>
    )
}
