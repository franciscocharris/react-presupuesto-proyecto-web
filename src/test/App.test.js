import React from 'react';
import { shallow } from "enzyme"
import { App } from "../App"

describe('pruebas en el componente padre App', () => {
    
    test('debe de renderizarse bien', () => {
        const wrapper = shallow(<App />);

        expect(wrapper).toMatchSnapshot();
    })
    
})
