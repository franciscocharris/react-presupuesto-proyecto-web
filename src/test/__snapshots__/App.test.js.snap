// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el componente padre App debe de renderizarse bien 1`] = `
<div
  className="container"
>
  <header>
    <h1>
      Gasto Semanal
    </h1>
    <a
      className="credito"
      href="https://www.facebook.com/profile.php?id=100013541311546"
      rel="noreferrer"
      target="_blank"
    >
       - por: Francisco M. Charris C. -
    </a>
    <div
      className="contenido-principal contenido"
    >
      <Pregunta
        setPresupuesto={[Function]}
        setRestante={[Function]}
        setShowPregunta={[Function]}
      />
    </div>
  </header>
</div>
`;
